pkg/plugins/plugins.go

```diff
@@ -251,6 +253,14 @@ func (p *Plugin) Logger() log.Logger {
 	return p.log
 }
 
+func (p *Plugin) Tracer() tracing.Tracer {
+	return p.trace
+}
+
+func (p *Plugin) SetTracer(t tracing.Tracer) {
```

Comment:
This doesn't seem to be called. Since the tracer is the same for all plugins, might be easier to skip `Plugin` holding a tracer and pass the tracer around as dependency to services that needs it for calling the `PluginFactoryFunc`?

Reply:
ok